### PR TITLE
Node id block startup if file exists

### PR DESCRIPTION
**Pull request by Caleb-Hill**
As node_id moves into the database eventually we want splinterd not to start with the old node_id storage file in place. This makes splinterd check for the file and exit with an error message if it exists.

### CODE REVIEW

**Review by peterschwarz — CHANGES_REQUESTED** [review]
Tested.  Works as expected when upgrading from the yaml file to a DB stored node-id. However, it does not have the existing behaviour of starting up and generating a node id if one is missing and not provided on the command line.

**Review by peterschwarz — APPROVED** [review]
Tested

**Review by peterschwarz — APPROVED** [review]

File: splinterd/src/main.rs [hunk 1]
```diff
@@ -97,65 +102,116 @@ fn create_config(_toml_path: Option<&str>, _matches: ArgMatches) -> Result<Confi
...
+                }
+                None => {
+                    // If the config object does not have a node_id, continue with the node_id read
+                    // from the file.
+                    Ok(file_node_id)
+                }
+            }
+        } else {
+            // If node file does not exist, need to create and save a node_id file.
+            // Check if the config obejct has a node_id, otherwise generate a random one.
```
Comment (isabeltomb) [inline splinterd/src/main.rs:187]: s/obejct/object

File: splinterd/src/main.rs [hunk 2]
```diff
@@ -97,65 +102,116 @@ fn create_config(_toml_path: Option<&str>, _matches: ArgMatches) -> Result<Confi
...
+                .database()
+                .parse()
+                .map_err(|_| UserError::InvalidArgument("db_connection".to_string()))?;
+            let store = create_store_factory(database_uri)?.get_node_id_store();
+            let db_node_id = store.get_node_id();
+            let config_node_id = config.node_id();
+            let save_new_node_id = |node_id| -> Result<(), UserError> {
+                store
+                    .set_node_id(node_id)
+                    .map_err(|err| UserError::from(InternalError::from_source(Box::new(err))))
```
Comment (isabeltomb) [inline splinterd/src/main.rs:124]: I think "node-file-block" needs to be added to the feature guard for `use splinter::error::InternalError;` on line 44, attempting to compile with just the `node-file-block` feature returns the error: "failed to resolve: use of undeclared type `InternalError`"

**Review by isabeltomb — COMMENTED** [review]
Tested

**Review by isabeltomb — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**